### PR TITLE
Redirect "/list" to "/" when there is no token

### LINKED ISSUE #24: 14. As a user, I want "/list" to redirect to the home page if there is no token
## Summary
If the user does not yet have a token, we don't want the `List` view to be visible. Instead, we want to redirect the user to the home page to create a new list or enter their token for an existing list. Once a token is stored in local storage, the user will then be able to view the `List` view.

## Acceptance Criteria
- [x] If the user doesn't already have a token, clicking `List` in the nav bar or entering the '/list' URL will redirect to the home page.

### PR DESCRIPTION
**Pull request by emilysellers**
## Description
This PR fixes the user being able to view the `List` view if they don't yet have a token. Now, if the user doesn't have a token and clicks `List` in the nav bar or enters "/list" in the URL bar, the `Home` view persists. Once the user creates a new list or enters a token for an existing list, the token is stored in local storage and the user is then able to view the `List` view.

Something to consider: should the same check be implemented for the `Add Item` view so that if a user doesn't yet have a token, they are not able to navigate to the `Add Item` view?

## Related Issue
Closes #24

## Acceptance Criteria
- [x]  If the user doesn't already have a token, clicking `List` in the nav bar or entering "/list" in the URL bar will keep the user on the home page.

## Type of Changes
|     | Type                       |
| --- | -------------------------- |
| ✓ | :bug: Bug fix              |
|     | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
In this before video, there is no token in local storage. The user clicks the `List` link in the nav bar and is redirected to the `List` view.

https://user-images.githubusercontent.com/84364905/234950851-604acb78-3740-4630-88b4-b3e17630d24b.mov

### After
In this after video, there is no token in local storage. The user clicks the `List` link in the nav bar and still sees the `Home` view, then enters "/list" in the URL bar and still sees the `Home` view. The user then clicks the `Create new list` button, the token is saved to local storage and the user is redirected to the `List` view.

https://user-images.githubusercontent.com/84364905/234951735-96213eb9-3754-464c-8831-36ab57331194.mov

## Testing Steps / QA Criteria
- Visit the preview URL for this PR: https://tcl-57-smart-shopping-list--pr25-es-no-token-redirect-k44q2dr8.web.app/
- Click on `List` in the nav bar and check that the `Home` view persists.
- Enter "/list" in the URL bar and check that the `Home` view persists.
- Click `Create new list` button. Check that a token is saved in local storage and that you're redirected to the `List` view.

### CODE REVIEW

**Review by djtaylor8 — APPROVED** [review]
Thanks for doing this, @emilysellers!

**Comment by Amy-Pr** [comment]
Thanks Emily! To answer your question, I would vote for doing the same with the Add Item view.

**Review by Amy-Pr — APPROVED** [review]
Thanks Emily! To answer your question, I would vote for doing the same with the Add Item view.

File: src/App.jsx [hunk 1]
```diff
@@ -66,7 +66,10 @@ export function App() {
 							)
 						}
 					/>
-					<Route path="/list" element={<List data={data} />} />
+					<Route
+						path="/list"
+						element={listToken ? <List data={data} /> : <Navigate to="/" />}
+					/>
```
Comment (danainjax) [inline src/App.jsx]: It's a very useful line of code, @emilysellers. Thank you for doing this extra to make the app more usable. 😁👍🏻

**Review by danainjax — APPROVED** [review]
Good & thanks!

**Pull request merged**